modules/core/src/main/scala/trace4cats/package.scala

```diff
@@ -0,0 +1,30 @@
+import cats.data.Kleisli
+import cats.effect.kernel.Resource
+import trace4cats.model.{SpanKind, TraceHeaders}
+import trace4cats.{kernel => t4ckernel}
+
+import scala.annotation.unchecked.uncheckedVariance
+
+package object trace4cats {
+  type SpanName = String
+  type SpanParams = (SpanName, SpanKind, TraceHeaders, ErrorHandler)
+  type ResourceKleisli[F[_], -In, +Out] = Kleisli[Resource[F, +*], In, Out @uncheckedVariance]
```

Comment:
I was going to suggest renaming `ResourceKleisli` to something that reflects purpose rather than implementation, and possibly changing it from a type alias to part of a class heirarchy with `EntryPoint`. But again, not something that should happen in this PR - just mentioning it while I remember

Reply:
The main reason for not making this its own class is that we get a lot of things for free with `Kleisli` and its input and output aren't fixed to the input/output of the entry point. This allows us to compose the entrypoint with other means of establishing some context from some input. There's an example of this in the[ http4s library](https://github.com/trace4cats/trace4cats-http4s/blob/master/modules/http4s-server/src/main/scala/io/janstenpickle/trace4cats/http4s/server/Http4sResourceKleislis.scala).